perfkitbenchmarker/benchmark_spec.py

```diff
@@ -229,7 +229,19 @@ def ConstructVirtualMachines(self):
 
         if DISK_SPEC in group_spec:
           disk_spec_class = disk.GetDiskSpecClass(cloud)
-          disk_spec = disk_spec_class(**group_spec[DISK_SPEC][cloud])
+          disk_spec_dict = group_spec[DISK_SPEC][cloud]
+          # We have to translate the data_disk_type flag here because
+          # this is where we know for sure what cloud we're using, and
+          # therefore we're able to pick the right translation table.
+          if FLAGS['data_disk_type'].present:
```

Comment:
A simpler flow for this would be doing this check after the flags get applied (on line 245).
You would only need one line (and wouldn't need to do all this flag presence checking):
disk_spec.disk_type = disk.WarnAndTranslateDiskTypes(disk_spec.disk_type, cloud)